hotrod/src/main/java/io/hyperfoil/hotrod/steps/HotRodRequestStep.java

```diff
@@ -0,0 +1,98 @@
+package io.hyperfoil.hotrod.steps;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+
+import io.hyperfoil.api.session.Access;
+import io.hyperfoil.api.session.ResourceUtilizer;
+import io.hyperfoil.api.session.Session;
+import io.hyperfoil.api.statistics.Statistics;
+import io.hyperfoil.core.builders.SLA;
+import io.hyperfoil.core.session.SessionFactory;
+import io.hyperfoil.core.steps.StatisticsStep;
+import io.hyperfoil.function.SerializableFunction;
+import io.hyperfoil.hotrod.api.HotRodConnectionPool;
+import io.hyperfoil.hotrod.api.HotRodDestinationTable;
+import io.hyperfoil.hotrod.api.HotRodOperation;
+import io.hyperfoil.hotrod.resource.HotRodResource;
+import io.hyperfoil.hotrod.resource.HotRodResourceKey;
+
+public class HotRodRequestStep extends StatisticsStep implements ResourceUtilizer, SLA.Provider {
+
+   final HotRodResourceKey futureWrapperKey;
+   final SerializableFunction<Session, HotRodOperation> operation;
+   final SerializableFunction<Session, String> cacheName;
+   final Access cacheKeyAccess;
+   final Access cacheValueAccess;
+
+   protected HotRodRequestStep(int id, HotRodResourceKey futureWrapperKey,
+                               SerializableFunction<Session, HotRodOperation> operation,
+                               SerializableFunction<Session, String> cacheName,
+                               Access cacheKeyAccess,
+                               Access cacheValueAccess) {
+      super(id);
+      this.futureWrapperKey = futureWrapperKey;
+      this.operation = operation;
+      this.cacheName = cacheName;
+      this.cacheKeyAccess = cacheKeyAccess;
+      this.cacheValueAccess = cacheValueAccess;
+   }
+
+   @Override
+   public SLA[] sla() {
+      return new SLA[0];
+   }
+
+   @Override
+   public boolean invoke(Session session) {
+
+      String cacheName = this.cacheName.apply(session);
+      HotRodOperation operation = this.operation.apply(session);
+
+      Object key = cacheKeyAccess.getObject(session);
+      Object value = cacheValueAccess.getObject(session);
+
+      HotRodDestinationTable destinations = HotRodDestinationTable.get(session);
+      HotRodConnectionPool pool = destinations.getPools().get(cacheName);
+      RemoteCache remoteCache = pool.getRemoteCache();
+
+      CompletableFuture future;
+      Statistics statistics = session.statistics(id(), HotRodMetric.METRIC_DURATION);
```

Comment:
`duration` is not a good name for metric - duration as for latency is just one aspect of the statistics.
Use session.currentSequence().definition().name() and delete the class `HotRodMetric`.

Reply:
Fixed